### PR TITLE
Refactor FXIOS-8143 [v123] Credit card and Logins Saved_all Probes not Triggering at Startup

### PR DESCRIPTION
**Pull request by petruSt24**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8143)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18127)

## :bulb: Description

Moved the telemetry events from their corresponding screens to app startup
## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [X] Wrote unit tests and/or ensured the tests suite is passing
- [X] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [X] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: firefox-ios/Client/Application/AppDelegate.swift
Comment (cyndichin) [inline firefox-ios/Client/Application/AppDelegate.swift:311]: thanks for looking into this! what do you think about adding this in its own class / struct to decouple from the AppDelegate? Similar to how we have `WebviewTelemetry`, `SponsoredTileTelemetry`, `PrivateBrowsingTelemetry`

**Comment by cyndichin** [comment]
also tagging @nbhasin2 for review

File: firefox-ios/Client/Application/AppDelegate.swift
Comment (nbhasin2) [inline firefox-ios/Client/Application/AppDelegate.swift:303]: @petruSt24 Few questions 
a) why do we want to send num of cc saved on each launch? 
b) also, why can't we do it as part of when browser view controller loads and why does it need to happen in AppDelegate?
c) If you are sending it on every launch then did you ask if we are to fully remove it from the cc page?

Lastly I don't feel comfortable bloating AppDelegate with these types of changes, can we not add a separate class or struct for a cleaner approach if we do have to do these changes?

File: firefox-ios/Client/Application/AppDelegate.swift
Comment (petruSt24) [reply firefox-ios/Client/Application/AppDelegate.swift:311]: can do, initially I thought that only 2 measly events won't clutter the class too much but I see how this can grow in time so I'll make it a separate class

**Review by cyndichin — APPROVED** [review]
thank you! looks good from my side :) will wait for @nbhasin2's approval as well before merging.

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**
